### PR TITLE
refactor: use ISet and ICState in interfaces

### PR DESCRIPTION
**Pull request by chad-js**
getting this working for integration with `cozy-v2-developer-guides`

### CODE REVIEW

File: src/interfaces/ICozyLens.sol [hunk 1]
```diff
@@ -1,29 +1,12 @@
...
-import "src/interfaces/IDecayModel.sol";
-import "src/interfaces/IDripModel.sol";
+import "src/interfaces/ISet.sol";
 
 /**
  * @notice Helper contract for reading data from the Cozy protocol.
  */
 interface ICozyLens {
-  /// @notice  Set-level configuration.
-  struct SetConfig {
```
Comment (chad-js) [inline src/interfaces/ICozyLens.sol:12]: inlining these structs here makes it weird for integrators - in some cases they'll have to cast `IConfig.SetConfig` to `ICozyLens.SetConfig` for example

File: src/interfaces/ICozyLens.sol [hunk 2]
```diff
@@ -44,76 +28,64 @@ interface ICozyLens {
...
-  function allTriggers(address _set) view external returns (address[] memory _triggers);
-
-  // -------------------------------------
-  // -------- Address Computation --------
-  // -------------------------------------
-
-  /// @notice Given the `_salt`, `_asset`, `_setConfig`, and `_marketInfos` that will be used to create a set,
-  /// compute and return the address that set will be deployed to.
-  /// @dev The `_salt` is the user-provided salt, not the final salt after hashing with the chain ID.
-  function computeSetAddress(bytes32 _salt, address _asset, SetConfig memory _setConfig, SetConfig[] memory _marketInfos) view external returns (address);
```
Comment (chad-js) [inline src/interfaces/ICozyLens.sol:115]: these two funcs have been removed from `CozyLens` and moved to the set and ptoken factories

**Review by Dendrimer — APPROVED** [review]

File: src/interfaces/ICozyLens.sol (hunk 1 above)
Comment (mds1) [reply src/interfaces/ICozyLens.sol:12]: Can they be defined outside of the interface? Can't recall offhand if that works for structs or just functions/errors. For example:

```solidity
// IWhatever.sol
struct X {
  uint256 a;
}

interface Y {
  function foo(X x) external;
}
```

Then when you import the above file, it'd be available directly

Comment (mds1) [reply src/interfaces/ICozyLens.sol:12]: Ah I see these live in `IConfig` anyway, but same concept: could potentially have no `interface {` there so you don't need `IConfig.SetConfig` in other files

**Review by davidlaprade — APPROVED** [review]

File: src/interfaces/ICozyLens.sol (hunk 1 above)
Comment (chad-js) [reply src/interfaces/ICozyLens.sol:12]: [2496369](https://github.com/Cozy-Finance/cozy-v2-interfaces/pull/7/commits/2496369daa4d572e2bdab0a3dceaa4b221b83a12) this seems to work well, I tested by locally updated the dependency in cozy-v2-triggers as well

**Pull request merged**